File: spring_java_orm_sample/src/main/java/com/example/demo/doma/entity/BookWithPublisherNameAggregateStrategy.java
package com.example.demo.doma.entity;

import org.seasar.doma.AggregateStrategy;
import org.seasar.doma.AssociationLinker;

import java.util.Objects;
import java.util.function.BiFunction;

@AggregateStrategy(root = BookWithPublisherName.class, tableAlias = "b")
public interface BookWithPublisherNameAggregateStrategy {
    @AssociationLinker(propertyPath = "bookStockList", tableAlias = "bs")
    BiFunction<BookWithPublisherName, BookStockWithStoreName, BookWithPublisherName> bookStockList =
        (book, bookStock) -> {
            if (Objects.nonNull(bookStock) && Objects.nonNull(bookStock.getId())) {
                book.getBookStockList().add(bookStock);
            }
            return book;
        };
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/doma/entity/BookWithPublisherNameAggregateStrategy.kt
package com.example.demo.doma.entity

import org.seasar.doma.AggregateStrategy
import org.seasar.doma.AssociationLinker
import java.util.*
import java.util.function.BiFunction

@AggregateStrategy(root = BookWithPublisherName::class, tableAlias = "b")
interface BookWithPublisherNameAggregateStrategy {
    companion object {
        @JvmField
        @AssociationLinker(propertyPath = "bookStockList", tableAlias = "bs")
        val bookStockList: BiFunction<BookWithPublisherName?, BookStockWithStoreName?, BookWithPublisherName?> =
            BiFunction { book: BookWithPublisherName?, bookStock: BookStockWithStoreName? ->
                if (Objects.nonNull(bookStock) && Objects.nonNull(bookStock!!.id)) {
                    book!!.bookStockList!!.add(bookStock)
                }
                book
            }
    }
}
